src/Indexer.js

```diff
@@ -86,6 +123,31 @@ export default class Indexer {
     return identifier
   }
 
+  /**
+   * Parses titles and subtitles out of a JSON body
+   * @param {Object} json - A Trellis resource (of some kind: RDFSource, BasicContainer, NonRDFSource, etc.)
+   * @returns {Object} an object containing title and subtitle strings if any found, null if not
+   */
+  titlesAndSubtitlesFrom(json) {
+    const titles = JSONPath({
+      json: json,
+      path: '$..mainTitle',
+      flatten: true
+    })
+      .filter(obj => obj['@value']) // Filter out hits without values, e.g., from the @context object
+      .map(obj => obj['@value']) // Extract the title value and ignore the @language for now
+
+    const subtitles = JSONPath({
+      json: json,
+      path: '$..subtitle',
+      flatten: true
+    })
+      .filter(obj => obj['@value']) // Filter out hits without values, e.g., from the @context object
+      .map(obj => obj['@value']) // Extract the title value and ignore the @language for now
+
+    return { title: titles, subtitle: subtitles }
```

Comment:
Worst-case scenario here is we find no titles and no subtitles, in which case we wind up passing empty arrays to the index() function above, which results in index entries for the IDs of the resources that were touched, but they will not be discoverable by title or subtitle (obviously).